specs/ServerCertificate.md

```diff
@@ -0,0 +1,289 @@
+
+  // Continues with the navigation to a server with a TLS certificate if
+  // the error status is of type `COREWEBVIEW2_WEB_ERROR_STATUS_CERTIFICATE_IS_INVALID`
+  // and trusted by the host app.
+  if (e.ErrorStatus == CoreWebView2WebErrorStatus.CertificateIsInvalid &&
+                            ValidateServerCertificate(certificate))
+  {
+    e.Handled = true;
+  }
+  else
+  {
+    // Cancel the request for other TLS certificate error types or if untrusted by the host app.
+    e.Cancel = true;
+  }
+}
+
+// Function to validate the server certificate for untrusted root or self-signed certificate.
+bool ValidateServerCertificate(CoreWebView2Certificate certificate)
+{
+    // Get the list of host app trusted certificates and its thumbprint.
+
+    // Get last chain element using `PemEncodedIssuerCertificateChain` property of `CoreWebView2Certificate`
+    // that contain raw data of untrusted root CA/self signed certificate. Get the untrusted
+    // root CA/self signed certificate thumbprint from the raw certificate data and
+    // validate thumbprint against the host app trusted certificate list.
+
+    // Return true if it exist in host app certificate trusted list, otherwise return false.
+    return true;
+}
+
+// This example clears the TLS decision in response to proceeding with TLS certificate errors.
+async void ClearServerCertificateErrorOverrideCache()
+{
+  await webView.CoreWebView2.ClearServerCertificateErrorOverrideCacheAsync();
+  MessageBox.Show(this, "Cleared", "Clear server certificate error override cache");
+}
+```
+
+# API Details
+
+## Win32 C++
+``` cpp
+[uuid(4B7FF0D2-8203-48B0-ACBF-ED9CFF82567A), object, pointer_default(unique)]
+interface ICoreWebView2_11 : ICoreWebView2_10 {
+  /// Add an event handler for the ServerCertificateError event.
+  /// The ServerCertificateError event is raised when the WebView2
+  /// cannot verify server's digital certificate while loading a web page.
+  ///
+  /// This event will raise only for top-level navigations as WebView2 block or cancel the
+  /// request for sub resources with server certificate error. Also, this event follows
+  /// the `NavigationStarting` event.
+  ///
+  /// With this event you have several options for responding to server
+  /// certificate error requests:
+  ///
+  /// Scenario                                                   | Handled | Cancel
+  /// ---------------------------------------------------------- | ------- | ------
+  /// Ignore the warning and continue the request                | True    | False
+  /// Cancel the request                                         | n/a     | True
+  /// Display default TLS interstitial page                      | False   | False
+  ///
+  /// If you don't handle the event, WebView2 will show the default TLS interstitial page to user.
+  ///
+  /// WebView2 caches the response when `Handled` is set `TRUE` for the host and certificate in the session
```

Comment:
Is this correct? We're talking about the host of the URI we're requesting that results in the server certificate with an error I guess right? And it works this way I guess because you could have different hosts using the same certificate?

```suggestion
  /// WebView2 caches the response when `Handled` is `TRUE` for the RequestUri's host and the server certificate in the session
```